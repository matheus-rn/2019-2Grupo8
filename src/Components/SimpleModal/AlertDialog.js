import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles';
import firebase from 'firebase';
import axios from 'axios';
import {success} from '../../Helpers/validates';
import {withRouter} from 'react-router-dom';


const theme = createMuiTheme({
    palette: {
      primary: { main: '#44a1f2' },
      secondary: { main: '#fafafa' },
    },
    typography: { useNextVariants: false },
    overrides: {
        MuiButton: {
          raisedPrimary: {
            color: 'white',
          },
        },
    },
  });

  
class  AlertDialog extends React.Component {

  state = { 
    open: false,
    setOpen: false,
  }
  
  delete_tutoring= ()=>{
    var token= {};
    var idTutoring = this.props.match.params.id_tutoring; 
      token["id_tutoring"] = idTutoring;
    firebase.auth().currentUser.getIdToken().then(function(idToken){
      token["access_token"] = idToken;
     
    });
  
    axios.post(process.env.REACT_APP_GATEWAY+"/delete_tutoring/", token)
        .then(res => {
          if(success(res)){
            this.props.history.push('/Feed');
          } 
        });
  }
  handleClickOpen = ()=> {
    this.setState({open: true});
  }

  handleClose = ()=>{
    this.setState({setOpen: false});
    this.props.history.push('/Feed');
  }
render(){
    return (
      <div>
          <MuiThemeProvider theme={theme}>
              <Button variant="contained" color="primary" onClick={this.handleClickOpen}>
                  excluir
              </Button>
          </MuiThemeProvider>
          <Dialog
          open={this.state.open}
          onClose={this.handleClose}
          aria-labelledby="alert-dialog-title"
          aria-describedby="alert-dialog-deion"
          >
              <DialogTitle id="alert-dialog-title">{"Excluir Monitoria?"}</DialogTitle>
              <DialogContent>
              <DialogContentText id="alert-dialog-deion">
                  Deseja realmente excluir a monitoria?
              </DialogContentText>
              </DialogContent>
              <DialogActions>
              <Button onClick={this.handleClose} color="primary">
                  cancelar
              </Button>
              <Button onClick={this.delete_tutoring} color="primary" autoFocus>
                  excluir
              </Button>
              </DialogActions>
        </Dialog>
      </div>
    );
  }
}

export default withRouter(AlertDialog);