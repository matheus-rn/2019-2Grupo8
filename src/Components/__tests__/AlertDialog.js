import React from 'react';
import AlertDialog from '../SimpleModal/AlertDialog';
import firebase from 'firebase';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import toJson from 'enzyme-to-json';

firebase.initializeApp({
  apiKey: '...',
  authDomain: '...',
  databaseURL: '...',
  projectId: '...',
  storageBucket: '...',
  messagingSenderId: '...',
});

let props={
    match:{
        params:{
            id_tutoring:jest.fn()    
    }}
};
const alertdialog= jest.fn();

describe('Testing AlertDialog component', () => {
    it('Test if AlertDialog renders correctly', () =>{
        jest.fn();
        const tree = shallow(<AlertDialog {...props}/>);
        expect(toJson(tree)).toMatchSnapshot();
    });
    it('Test if AlertDialog renders correctly', () =>{
      const tree = renderer.create(
        <Router><AlertDialog alertdialog={alertdialog}/></Router>
        )
        expect(toJson(tree)).toMatchSnapshot()
    });
});