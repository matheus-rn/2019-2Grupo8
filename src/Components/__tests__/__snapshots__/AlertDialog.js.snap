// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AlertDialog component Test if AlertDialog renders correctly 1`] = `
<Route>
  <Component />
</Route>
`;

exports[`Testing AlertDialog component Test if AlertDialog renders correctly 2`] = `undefined`;
